Jenkinsfile

```diff
@@ -106,44 +132,94 @@ pipeline {
+        }
+
+        stage('determine release tag') {
+            when {
+                tag pattern: RELEASE_TAG_REGEX, comparator: 'REGEXP'
+            }
 
+            steps {
                 script {
-                    def properties = readProperties(file: 'version.properties')
-                    version = "${properties.version}-${currentBuild.startTimeInMillis}.${currentBuild.number}"
-                    def images = [
-                        "\$IMAGE_REGISTRY/\$IMAGE_REPOSITORY:$version",
-                        "\$IMAGE_REGISTRY/\$IMAGE_REPOSITORY:latest"
-                    ]
-
-                    for (String image : images) {
-                        sh "docker tag \$IMAGE_REPOSITORY:latest $image"
-                        sh "docker push $image"
+                    tagVersion = sh (returnStdout: true, script: "git tag --points-at HEAD").trim()
+                }
+            }
+        }
+
+        stage('create github release') {
+            when {
+                anyOf {
+                    branch 'master'
+                    tag pattern: RELEASE_TAG_REGEX, comparator: 'REGEXP'
+                }
+            }
+
+            steps {
+                withCredentials([usernamePassword(credentialsId: 'github-token', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                    script {
+                        def createReleaseResponse = sh (
+                                returnStdout: true,
+                                script:
+                                        """
+                            curl -u '${GIT_USERNAME}:${GIT_PASSWORD}' \
+                                 -H 'Content-Type:application/json' \
+                                 'https://api.github.com/repos/${GITHUB_REPO_PATH}/releases' \
+                                 -d '{
+                                         "tag_name": "${tagVersion}",
+                                         "target_commitish": "$GIT_COMMIT",
+                                         "name": "${tagVersion}",
+                                         "prerelease": ${tagVersion ==~ RC_TAG_REGEX}
+                                     }'
+                            """
+                        )
                     }
                 }
             }
         }
 
+        stage('push image') {
+            when {
+                anyOf {
+                    branch 'master'
+                    tag pattern: RELEASE_TAG_REGEX, comparator: 'REGEXP'
+                }
+            }
+
+            steps {
+                withCredentials([usernamePassword(credentialsId: "$IMAGE_CREDS_JENKINS_ID", usernameVariable: 'IMAGE_REGISTRY_USERNAME', passwordVariable: 'IMAGE_REGISTRY_PASSWORD')]) {
+                    sh 'docker login $IMAGE_REGISTRY --username $IMAGE_REGISTRY_USERNAME --password $IMAGE_REGISTRY_PASSWORD'
+                    sh "docker tag \$IMAGE_REPOSITORY:latest \$IMAGE_REGISTRY/\$IMAGE_REPOSITORY:${tagVersion}"
```

Comment:
Is it possible to use the Docker API instead here?

Reply:
The Jenkins Docker API, i.e., the `docker.image(...).push()` stuff.

Reply:
Done :+1:

Reply:
Alas, due to this bug: https://issues.jenkins-ci.org/browse/JENKINS-44609 it's not possible with `Dockerfile.jvm` in its current form.

Until that bug is resolved, there are only two options:

1. Continue to use `sh 'docker...`; or
2. Refactor `Dockerfile.jvm` to remove the `AS <aslias>` parts from the `FROM` lines, and reference the different stages by index (e.g. `COPY --from=0 ...`).